instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/servlet/ServletSpanNaming.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.servlet;
+
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.ContextKey;
+
+/**
+ * Helper container for tracking whether servlet integration should update server span name or not.
+ */
+public class ServletSpanNaming {
+
+  private static final ContextKey<ServletSpanNaming> CONTEXT_KEY =
+      ContextKey.named("opentelemetry-servlet-span-naming-key");
+
+  public static Context init(Context context) {
+    if (context.get(CONTEXT_KEY) != null) {
+      return context;
+    }
+    return context.with(CONTEXT_KEY, new ServletSpanNaming());
+  }
+
+  private volatile boolean servletUpdatedServerSpanName = false;
+
+  private ServletSpanNaming() {}
+
+  /**
+   * Returns true, if servlet integration should update server span name. After server span name has
+   * been updated with <code>setServletUpdatedServerSpanName</code> this method will return <code>
+   * false</code>.
+   *
+   * @param context server context
+   * @return <code>true</code>, if the server span name should be updated by servlet integration, or
+   *     <code>false</code> otherwise.
+   */
+  public static boolean shouldUpdateServerSpanName(Context context) {
```

Comment:
It looks like we lost the change I made to use CAS - are you sure you don't want to use it here @laurit? I think it makes the safety of the code more obvious, and less code.

Reply:
I changed the logic a bit. Previously span name was always updated when `shouldUpdateServerSpanName` was called, now we call `shouldUpdateServerSpanName`, if name was not already updated we generate new name and only if new name wasn't null (null means that we are currently executing a filter from which we can't get a good name) update span name. With using cas would need to generate a new span name and verify it is not null before calling `shouldUpdateServerSpanName` which I didn't want to do.
`shouldUpdateServerSpanName` is also called from integrations that set server span name from filter and need to suppress naming from servlet so that the name given from framework integration wouldn't be overwritten.
Anyway It is likely that we need to change how this works, probably need to add some kind of priorities to ensure that frameworks don't overwrite names given by each other. Currently for example when calling controller we set server span name to `/foo` when this controller throws an exception framework calls another controller which is mapped to `/error` to handle the exception. On call to second controller we set server span name to `/error` although it should remain as `/foo`. Also current approach isn't well suited for portals where you could have multiple frameworks with their controllers sitting on the same page each trying to have the final say in what the span name should be.